### DISCUSSION TITLE
Can we support ChatPromptTemplate for function import_semantic_plugin_from_directory()?

### DISCUSSION THREAD

**Discussion by HuskyDanny** [microsoft/semantic-kernel#4692] [Q&A]
The function below can only register function with promptTemplate, can we extend this to register function with ChatPromptTemplate ? I need some skills to have a history of messages, and want to govern them at the same place in the same way.

    def import_semantic_plugin_from_directory(
        self, parent_directory: str, plugin_directory_name: str
    ) -> Dict[str, SKFunctionBase]:
        CONFIG_FILE = "config.json"
        PROMPT_FILE = "skprompt.txt"

        validate_plugin_name(plugin_directory_name)

        plugin_directory = os.path.join(parent_directory, plugin_directory_name)
        plugin_directory = os.path.abspath(plugin_directory)

        if not os.path.exists(plugin_directory):
            raise ValueError(f"Plugin directory does not exist: {plugin_directory_name}")

        plugin = {}

        directories = glob.glob(plugin_directory + "/*/")
        for directory in directories:
            dir_name = os.path.dirname(directory)
            function_name = os.path.basename(dir_name)
            prompt_path = os.path.join(directory, PROMPT_FILE)

            # Continue only if the prompt template exists
            if not os.path.exists(prompt_path):
                continue

            config_path = os.path.join(directory, CONFIG_FILE)
            with open(config_path, "r") as config_file:
                config = PromptTemplateConfig.from_json(config_file.read())

            # Load Prompt Template
            with open(prompt_path, "r") as prompt_file:
                template = PromptTemplate(prompt_file.read(), self.prompt_template_engine, config)

            # Prepare lambda wrapping AI logic
            function_config = SemanticFunctionConfig(config, template)

            plugin[function_name] = self.register_semantic_function(
                plugin_directory_name, function_name, function_config
            )

        return plugin

**Comment by markwallace-microsoft** [comment]
I am cleaning up old unanswered discussions
